File: py/lm.py
test=int(raw_input())

v=1000000007

def comb(n,k):
	i=1
	p=1
	if(k>n-k):
		k=n-k;
			
	for i in range(1,k+1):
		p*=n
		n=n-1
		p/=i
	return p
	

while test>0:
	a=int(raw_input())
	val=0
	
	for i in range(1,a+1):
		val+=((((comb(a,i)%v)**2)%v)*i)%v
	
	print val
	
	test=test-1

File: c,c++/lm.c
#include<stdio.h>
#include<math.h>

long long int a[1000002]={0};

int v=1000000007;

long long int comb(int n)
{
	long long int i,p;
	i=1;
	p=a[n];
			
	if(p==0)
		a[n]=(2*(2*n-1)*comb(n-1)/(n-1));
	else
		return p;
		
	if(a[n]>v)
		a[n]=a[n]%v;
		
	p=a[n];
	return p;
}
	
int main()
{	
	a[1]=1;
	a[2]=6;
	int test;
	scanf("%d",&test);
	while(test--)
	{
		int a,i;
		scanf("%d",&a);
		long long int val=0;
		
		val+=comb(a);
		
		printf("%lld\n",val);
	}
	
	return 0;
}
